### PR TITLE
Rename @osdk/omniapi to @osdk/foundry

### PR DESCRIPTION
**Pull request by ericanderson**
- Renames the package(s)
- Marks @osdk/foundry as not private
- Makes some of the packages ESM only:
  - "@osdk/foundry",
  - "@osdk/client",
  - "@osdk/cli.*",
  - "@osdk/cli",
  - and some internal packages

### CODE REVIEW

File: .monorepolint.config.mjs [hunk 1]
```diff
@@ -45,6 +45,11 @@ const legacyPackages = [
   "@osdk/shared.net",
 ];
 
+const esmOnlyPackages = [
+  "@osdk/foundry",
+  "@osdk/foundry-generator",
+];
+
```
Comment (ericanderson) [inline .monorepolint.config.mjs:52]: I want to start pushing the non 1.0 stuff to be esm only.

File: monorepo/mytsup/tsup.mjs [hunk 2]
```diff
@@ -1,24 +1,27 @@
...
+ * @param {{cjsExtension?: ".cjs" | ".js", esmOnly?: boolean }} ourOptions
+ * @returns {Promise<import('tsup').Options>}
  */
 export default async (options, ourOptions) => {
   const babel = (await import("esbuild-plugin-babel")).default;
 
-  const packageJson = await readFile("package.json").then(f => JSON.parse(f));
+  const packageJson = await readFile("package.json", "utf-8").then(f =>
+    JSON.parse(f)
+  );
```
Comment (ericanderson) [inline monorepo/mytsup/tsup.mjs:16]: "bug" caught by adding `// @ts-check`. Wasnt reading as utf8

File: monorepo/mytsup/tsup.mjs [hunk 3]
```diff
@@ -1,24 +1,27 @@
...
 
   return {
     entry: [
       "src/index.ts",
       "src/public/*.ts",
       "src/public/*.mts",
     ],
-    format: ["cjs", "esm"],
+    format: ourOptions?.esmOnly ? ["esm"] : ["cjs", "esm"],
```
Comment (ericanderson) [inline monorepo/mytsup/tsup.mjs:24]: Minor build time improvement.

File: monorepo/mytsup/tsup.mjs [hunk 4]
```diff
@@ -39,25 +42,28 @@ export default async (options, ourOptions) => {
     sourcemap: true,
     splitting: true,
     minify: false, // !options.watch,
-    onSuccess: () => {
+    onSuccess: async () => {
```
Comment (ericanderson) [inline monorepo/mytsup/tsup.mjs:45]: "bug" caught by adding `// @ts-check`

File: monorepo/mytsup/tsup.mjs [hunk 5]
```diff
@@ -39,25 +42,28 @@ export default async (options, ourOptions) => {
...
 };
 
 async function readPackageVersion(k) {
-  const workspaceRoot = path.dirname(await findUp("pnpm-workspace.yaml"));
-  return await readFile(path.join(workspaceRoot, k, "package.json")).then(f =>
-    JSON.parse(f).version
-  );
+  const workspaceFile = await findUp("pnpm-workspace.yaml");
+  if (!workspaceFile) throw "couldnt find workspace file";
+  const workspaceRoot = path.dirname(workspaceFile);
```
Comment (ericanderson) [inline monorepo/mytsup/tsup.mjs:66]: "bug" caught by adding `// @ts-check`.

File: packages/foundry-generator/CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# @osdk/foundry-generator
```
Comment (ericanderson) [inline packages/foundry-generator/CHANGELOG.md:1]: this was just a rename and a move but git thinks its a new file

File: packages/client/CHANGELOG.md [hunk 7]
```diff
@@ -12,7 +12,7 @@
   - Fixes issues with where clauses for equality in subscriptions
   - Fixes issues with inconsistent idname and apiName's in the mapping data
 
-- 9906a41: Omniapi support
+- 9906a41: Foundry Platform support
```
Comment (styu) [inline packages/client/CHANGELOG.md:15]:
```suggestion
- 9906a41: Foundry Platform API support
```

**Review by styu — APPROVED** [review]

**Pull request merged**